tests/desktop/test_users_account.py

```diff
@@ -86,13 +86,21 @@ def test_hide_email_checkbox_works(self, mozwebqa):
         initial_state = view_profile_page.is_email_field_present
 
         edit_profile_page = home_page.header.click_edit_profile()
-        edit_profile_page.change_hide_email_state()
-        edit_profile_page.click_update_account()
+        # make sure email checkbox isn't already selected
+        if edit_profile_page.checkbox_status == True:
+            edit_profile_page.show_email_address()
+            edit_profile_page.click_update_account()
+            home_page.header.click_edit_profile()
+            edit_profile_page.hide_email_address()
+
+        else:
+            edit_profile_page.hide_email_address()
 
+        edit_profile_page.click_update_account()
         view_profile_page = home_page.header.click_view_profile()
         final_state = view_profile_page.is_email_field_present
 
-        Assert.not_equal(initial_state, final_state, 'The initial and final states are the same. The profile change failed.')
+        Assert.equal(view_profile_page.is_email_field_present, False, 'The email address checkbox is not hidden')
```

Comment:
I think the failure here is "The email address is not hidden", not "The email address *checkbox* is not hidden".